src/Std/Data/HashMap/Basic.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Std.Data.DHashMap.Basic
+
+set_option linter.missingDocs true
+set_option autoImplicit false
+
+/-!
+# Hash maps
+
+This file develops the type `Std.Data.HashMap` of dependent hash maps.
+
+The operations `map` and `filterMap` on `Std.Data.HashMap` are defined in the module
+`Std.Data.HashMap.AdditionalOperations`.
+
+Lemmas about the operations on `Std.Data.HashMap` are available in the
+module `Std.Data.HashMap.Lemmas`.
+
+See the module `Std.Data.HashMap.Raw` for a variant of this type which is safe to use in
+nested inductive types.
+-/
+
+universe u v w
+
+variable {α : Type u} {β : Type v}
+
+namespace Std
+
+/-- Hash maps. -/
+structure HashMap (α : Type u) (β : Type v) [BEq α] [Hashable α] where
+  /-- Internal implementation detail of the hash map -/
+  inner : DHashMap α (fun _ => β)
+
+namespace HashMap
+
+@[inline, inherit_doc DHashMap.empty] def empty [BEq α] [Hashable α] (capacity := 8) :
+    HashMap α β :=
+  ⟨DHashMap.empty capacity⟩
+
+instance [BEq α] [Hashable α] : EmptyCollection (HashMap α β) where
+  emptyCollection := empty
+
+@[inline, inherit_doc DHashMap.insert] def insert [BEq α] [Hashable α] (m : HashMap α β) (a : α)
+    (b : β) : HashMap α β :=
+  ⟨m.inner.insert a b⟩
+
+@[inline, inherit_doc DHashMap.insertIfNew] def insertIfNew [BEq α] [Hashable α] (m : HashMap α β)
+    (a : α) (b : β) : HashMap α β :=
+  ⟨m.inner.insertIfNew a b⟩
+
+@[inline, inherit_doc DHashMap.containsThenInsert] def containsThenInsert [BEq α] [Hashable α]
+    (m : HashMap α β) (a : α) (b : β) : Bool × HashMap α β :=
+  let ⟨replaced, r⟩ := m.inner.containsThenInsert a b
+  ⟨replaced, ⟨r⟩⟩
+
+@[inline, inherit_doc DHashMap.containsThenInsertIfNew] def containsThenInsertIfNew [BEq α]
+    [Hashable α] (m : HashMap α β) (a : α) (b : β) : Bool × HashMap α β :=
+  let ⟨replaced, r⟩ := m.inner.containsThenInsertIfNew a b
+  ⟨replaced, ⟨r⟩⟩
+
+/-- Equivalent to (but potentially faster than) calling `get?` followed by `insertIfNew`. -/
+@[inline] def getThenInsertIfNew? [BEq α] [Hashable α] (m : HashMap α β) (a : α) (b : β) :
```

Comment:
Why is this not `@[inherit_doc]`?

Reply:
Ah, that would be because the docstring for `DHashMap.getThenInsertIfNew?` would talk about the `[LawfulBEq]` instance and the docstring for `DHashMap.Const.getThenInsertIfNew?` would talk about the `Const` prefix which no longer exists at the `HashMap` layer. I forgot to mention the `LawfulBEq` instance in the `DHashMap.getThenInsertIfNew?` docstring, I will fix that.

Reply:
Makes sense :)